src/js/facility-locator/components/Select.jsx

```diff
@@ -0,0 +1,160 @@
+  }
+
+  handleFilterSelect = (newOption, type) => {
+    const { currentQuery } = this.props;
+    const { facilityType, serviceType } = currentQuery; // eslint-disable-line no-unused-vars
+    if (type === 'facility' && newOption !== facilityType) {
+      if (newOption === 'AllFacilities') {
+        this.props.updateSearchQuery({
+          facilityType: null,
+          serviceType: null
+        });
+      } else {
+        this.props.updateSearchQuery({
+          facilityType: newOption,
+          serviceType: null
+        });
+      }
+    }
+    if (type === 'service' && newOption !== serviceType) {
+      if (newOption === 'All') {
+        this.props.updateSearchQuery({
+          serviceType: null,
+        });
+      } else {
+        this.props.updateSearchQuery({
+          serviceType: newOption
+        });
+      }
+    }
+    return this.toggleDropdown(null, type);
+  }
+
+  navigateDropdown = (e) => {
+    const type = this.props.optionType;
+    const isActive = this.state.dropdownActive;
+    const which = e.target;
+    if (isToggle(e, isActive)) {
+      if (isActive && isSelect(e.keyCode)) {
+        this.handleFilterSelect(which.id, type);
+      }
+      return this.toggleDropdown(e, type);
+    }
+    if (isActive && isTraverse(e)) {
+      const increment = getDirection(e.keyCode);
+      const newIndex = this.state.focusedIndex + increment;
+      return this.focusOption(this.options[newIndex], newIndex, type);
+    }
+    return false;
+  }
+
+  handleBlur = (e) => {
+    const { currentTarget } = e;
+    setTimeout(() => {
+      if (!currentTarget.contains(document.activeElement)) {
+        this.setState({ dropdownActive: false });
+      }
+    });
+  }
+
+  toggleDropdown = (e) => {
+    const type = this.props.optionType;
+    const isActive = this.state.dropdownActive;
+    const { currentQuery: { facilityType, serviceType } } = this.props; // eslint-disable-line no-unused-vars
+    const queryType = type === 'service' ? serviceType : facilityType;
+    if (type === 'service' && noServices(type, facilityType)) {
+      return;
+    }
+    // To handle unrelated key events
+    if (e) {
+      const shouldNotToggle = e.keyCode && !isToggle(e, isActive);
+      if (shouldNotToggle) return;
+    }
+    if (isActive) {
+      this.dropdown.focus();
+    } else {
+      const { selection, id } = getSelection(this.options, queryType);
+      this.focusOption(selection, id, type);
+    }
+    this.setState({
+      dropdownActive: !this.state.dropdownActive
+    });
+  }
+
+  focusOption = (option, index) => {
+    if (option) {
+      option.focus();
+    }
+    if (this.options[index]) {
+      this.setState({
+        focusedIndex: index
+      });
+    }
+  }
+
+  render() {
+    const { optionType, currentQuery, isMobile, hasIcons } = this.props;
+    const { facilityType, serviceType } = currentQuery;
+    const disabled = optionType === 'service' && !['benefits', 'vet_center'].includes(facilityType);
```

Comment:
Could we move this into the facility locator component? I think it may make things clearer if we get as much of the optionType switching out of this component and passed in via props.

Reply:
Good idea.

Reply:
Actually there is no switching happening here, this line checks whether the dropdown is disabled, originally for styling purposes, but now also to determine the disabled attribute on the button in light of your recommendation.